src/grpc_server.rs

```diff
@@ -0,0 +1,287 @@
+            }
+            Err(e) => {
+                return Err(Status::internal(format!(
+                    "Failed to delete application {}: {}",
+                    app.project_id, e
+                )));
+            }
+        };
+
+        //TODO: Delete the app from the database & prune its image
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
+    }
+
+    /// # Stop Deployment
+    /// Stop a deployment from its uuid.
+    /// # Arguments
+    /// The request containing the uuid of the deployment to stop.
+    /// # Returns
+    /// Nothing, wrapped in a Result.
+    async fn stop_deployment(
+        &self,
+        request: Request<StopDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let deployment_uuid = request.into_inner().deployment_uuid;
+
+        let app = match self.db.get_app(deployment_uuid.clone()) {
+            Ok(app) => app,
+            Err(e) => {
+                return Err(Status::not_found(e.to_string()));
+            }
+        };
+
+        let message: String = match self
+            .docker_engine
+            .stop_application(app.project_id.as_str(), app.application_id.as_str())
+            .await
+        {
+            Ok(_) => {
+                format!("Stopped application {}", app.project_id)
+            }
+            Err(e) => {
+                return Err(Status::internal(format!(
+                    "Failed to stop application {}: {}",
+                    app.project_id, e
+                )));
+            }
+        };
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
+    }
+
+    /// # Deployment Status
+    /// Get the status of a deployment from its uuid.
+    /// # Arguments
+    /// The request containing the uuid of the deployment to get the status of.
+    /// # Returns
+    /// Nothing, wrapped in a Result.
+    async fn deployment_status(
+        &self,
+        request: Request<DeploymentStatusRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let deployment_uuid = request.into_inner().deployment_uuid;
+
+        let app = match self.db.get_app(deployment_uuid.clone()) {
+            Ok(app) => app,
+            Err(e) => {
+                return Err(Status::not_found(e.to_string()));
+            }
+        };
+
+        let message = match self
+            .docker_engine
+            .get_application_status(app.project_id.as_str(), app.application_id.as_str())
+            .await
+        {
+            Ok(status) => format!("Application {} is {:?}", app.project_id, status),
+            Err(e) => {
+                return Err(Status::internal(format!(
+                    "Failed to get status of application {}: {}",
+                    app.project_id, e
+                )));
+            }
+        };
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
```

Comment:
You should also send the status separately from the message to avoid a gRPC client from having to parse the message after calling this function.
The `.proto` needs to be changed.